src/store/root.reducer.ts

```diff
@@ -1,21 +1,12 @@
-const initiatState = {
-  data: {},
-  petsToCompare: [],
-  search: {},
-};
+import { createReducer } from 'typesafe-actions';
 
-export const rootReducer = (state: object = initiatState, action: any) => {
-  switch (action.type) {
-    case 'PUT_PETS':
-      return {
-        ...state,
-        petsToCompare: action.payload.pets,
-      };
-    case 'SEARCH':
-      return {
-        search: action.payload,
-      };
-  }
+type State = {} | [];
```

Comment:
What is the reason to have state as array and object at the same time? Is there any option to not use different types sententiously? It treats as "code smell" if it used with no reason.